### ISSUE TITLE
HTTP Accept headers are not parsed correctly

### ISSUE THREAD

**Issue by k1rill-fedoseev** [sigp/lighthouse#3114]
## Description

Endpoint `/eth/v2/debug/beacon/states/<state_id>` does not correctly handle complex format of the Accept HTTP header.

Nimbus client has a similar functionality of checkpoint sync, during which it requests the latest state from the reliable Beacon Node API. However, the such request failes due to the following error:
```
{"code":400,"message":"BAD_REQUEST: Unsupported endpoint version: v2","stacktraces":[]}
```

It appears that the problem is in the Accept header format that Nimbus client provides along with the request: `Accept: application/octet-stream,application/json;q=0.9`, which Lighthouse cannot parse correctly.

https://github.com/status-im/nimbus-eth2/blob/13b264d509d48909bb354f85b879f22641723509/beacon_chain/spec/eth2_apis/rest_types.nim#L30

## Version

All up to v2.1.5

## Present Behaviour

Behaves like the endpoint does not exists.

## Expected Behaviour

Should return an SSZ-formatted state, as requested by the Accept header.

## Steps to resolve

For example, the issue is observed on the following public GBC Beacon Api endpoint, backed by the v2.1.5 Lighthouse nodes.
```
$ curl -H 'Accept: application/octet-stream,application/json;q=0.9' https://rpc-gbc.gnosischain.com/eth/v2/debug/beacon/states/head
{"code":400,"message":"BAD_REQUEST: Unsupported endpoint version: v2","stacktraces":[]}
$ curl -H 'Accept: application/octet-stream' https://rpc-gbc.gnosischain.com/eth/v2/debug/beacon/states/head > state.ssz
```

**Comment by michaelsproul** [comment]
Thanks for reporting!

This is a deficiency in our HTTP header parsing whereby we're parsing Accept headers in a very limited way, and ignoring the more complex Accept parsing defined in RFC7231: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2.

It seems our HTTP library `warp` doesn't provide any functionality for parsing Accept headers, so we may need to do a bit of leg work.

I found what looks like a great implementation in `actix_web` here: https://actix.rs/actix-web/actix_web/http/header/struct.Accept.html. Perhaps it would be possible to write some translation glue code in order to use that.

Another option would be to hand-roll our own implementation or create a crate specifically for this purpose to contribute back to the `warp` community. The `mime` crate does most of the heavy lifting, but lacks the comma splitting and q-factor prioritisation. I've sketched the beginning of what this might look like here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bf69044d030d03df6da87eb6c1b4d61e. Although we shouldn't use `.split(',')` due to the presence of quoted strings (see: https://github.com/http-rs/http-types/issues/349).
